src/SelfServe/SelfServeComponent.tsx

```diff
@@ -311,34 +313,41 @@ export class SelfServeComponent extends React.Component<SelfServeComponentProps,
     this.performSave();
   };
 
-  public isDiscardButtonDisabled = (): boolean => {
-    if (this.state.isSaving) {
-      return true;
-    }
+  public isInputModified = (): boolean => {
     for (const key of this.state.currentValues.keys()) {
-      const currentValue = JSON.stringify(this.state.currentValues.get(key));
-      const baselineValue = JSON.stringify(this.state.baselineValues.get(key));
+      const currentValue = this.state.currentValues.get(key);
+      if (currentValue && currentValue.hidden === undefined) {
```

Comment:
[nit] currentValue?.hidden === undefined

Reply:
problem is if currentvalue is undefined, currentValue?.hidden === undefined returns true. I want currentValue to be present and the hidden prop to be undefined.